packages/bolt-connection/src/bolt/bolt-protocol-v5x1.js

```diff
@@ -0,0 +1,132 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import BoltProtocolV5x0 from './bolt-protocol-v5x0'
+
+import transformersFactories from './bolt-protocol-v5x1.transformer'
+import Transformer from './transformer'
+import RequestMessage from './request-message'
+import { LoginObserver, LogoffObserver } from './stream-observers'
+
+import { internal } from 'neo4j-driver-core'
+
+const {
+  constants: { BOLT_PROTOCOL_V5_1 }
+} = internal
+
+export default class BoltProtocol extends BoltProtocolV5x0 {
+  get version () {
+    return BOLT_PROTOCOL_V5_1
+  }
+
+  get transformer () {
+    if (this._transformer === undefined) {
+      this._transformer = new Transformer(Object.values(transformersFactories).map(create => create(this._config, this._log)))
+    }
+    return this._transformer
+  }
+
+  get supportsLogoff () {
+    return true
+  }
+
+  /**
+   * Initialize a connection with the server
+   *
+   * @param {Object} param0 The params
+   * @param {string} param0.userAgent The user agent
+   * @param {any} param0.authToken The auth token
+   * @param {function(error)} param0.onError On error callback
+   * @param {function(onComplte)} param0.onComplete On complete callback
+   * @returns {LoginObserver} The Login observer
+   */
+  initialize ({ userAgent, authToken, onError, onComplete } = {}) {
+    const state = {}
+    const observer = new LoginObserver({
+      onError: error => this._onLoginError(error, onError),
+      onCompleted: metadata => {
+        state.metadata = metadata
+        return this._onLoginCompleted(metadata)
+      }
+    })
+
+    this.write(
+      RequestMessage.hello5x1(userAgent, this._serversideRouting),
+      observer,
+      false
+    )
+
+    return this.login({
+      authToken,
+      onComplete: metadata => onComplete({ ...metadata, ...state.metadata }),
+      onError,
+      flush: true
+    })
+  }
+
+  /**
+   * Performs login of the underlying connection
+   *
+   * @param {Object} param
+   * @param {Object} param.authToken the authentication token.
+   * @param {function(err: Error)} param.onError the callback to invoke on error.
+   * @param {function()} param.onComplete the callback to invoke on completion.
+   * @param {boolean} param.flush whether to flush the buffered messages.
+   *
+   * @returns {StreamObserver} the stream observer that monitors the corresponding server response.
+   */
+  login ({ authToken, onComplete, onError, flush } = {}) {
```

Comment:
The Server Calls this Logon, i don't know what the other drivers do and if it should be logon or login but i suspect either the server should change or the drivers so we all align nicely

Reply:
I've started with login because the driver already uses observer called `LoginObserver`, but it is fair enough keep the message with same name as in the server since it will make easier to people read the logs and follow the code, for example.